### ISSUE TITLE
fix: cycle detection in dependencyMapper — use visited set pattern

### ISSUE THREAD

**Issue by YusukeHirao** [markuplint/markuplint#3336]
## Parent Issue

Sub-issue of #3335

## Problem

The current cycle detection in `dependency-mapper.ts` only checks if the current element name matches the **start node** (`identifier`). This means intermediate cycles like A→B→C→B are not detected.

```typescript
// Current (line 74): only detects A→...→A
if (elName === identifier) {
    via.push('...[Recursive]');
    break;
}
```

## Solution

Replace with the standard DFS visited set pattern:

```typescript
const visited = new Set<string>();

while (true) {
    const mappedPretender = map.get(elName);
    if (!mappedPretender) break;

    if (visited.has(elName)) {
        via.push('...[Recursive]');
        break;
    }
    visited.add(elName);

    identity = mappedPretender[0];
    filePath = mappedPretender[1];
    via.push(elName);
    elName = getElName(identity);
}
```

## Scope

- File: `packages/@markuplint/pretenders/src/dependency-mapper.ts`
- Add test case for intermediate cycle detection (A→B→C→B)
- No API changes, no type changes

## Dependencies

None — can be implemented immediately.
